### PR TITLE
Documenting that WorkerContext is threadsafe

### PR DESCRIPTION
**Pull request by gokulavasan**

### CODE REVIEW

**Comment by gokulavasan** [comment]
@ParthGandhi @awholegunch Please review.

File: cdap-docs/developers-manual/source/building-blocks/workers.rst [hunk 1]
```diff
@@ -52,7 +52,9 @@ Their instances may be updated via the :ref:`Command Line Interface <cli-availab
     }
   }
 
-Workers can access and use ``Dataset``\s via a ``DatasetContext`` inside their ``run`` method::
+Workers can access and use ``Dataset``\s via a ``DatasetContext`` inside the ``run`` method of a ``TxRunnable``.
+A ``TxRunnable`` can be executed using the ``execute`` method of ``WorkerContext``. Note that ``WorkerContext``
+is thread-safe and each thread will get its own instance of the ``Dataset`` being accessed::
```
Comment (awholegunch) [inline cdap-docs/developers-manual/source/building-blocks/workers.rst]: each thread will receive its own...

**Comment by awholegunch** [comment]
One comment; LGTM.

**Comment by gokulavasan** [comment]
@ParthGandhi @awholegunch Added more content. Please take a look.

**Comment by ParthGandhi** [comment]
Content LGTM. I'll let JJ review the language.

File: cdap-docs/developers-manual/source/building-blocks/workers.rst [hunk 2]
```diff
@@ -52,7 +52,11 @@ Their instances may be updated via the :ref:`Command Line Interface <cli-availab
     }
   }
 
-Workers can access and use ``Dataset``\s via a ``DatasetContext`` inside their ``run`` method::
+Workers can access and use ``Dataset``\s via a ``DatasetContext`` inside the ``run`` method of a ``TxRunnable``.
+A ``TxRunnable`` can be executed using the ``execute`` method of ``WorkerContext``. Note that ``WorkerContext``
```
Comment (awholegunch) [inline cdap-docs/developers-manual/source/building-blocks/workers.rst]: method of a ``WorkerContext``.

File: cdap-docs/developers-manual/source/building-blocks/workers.rst [hunk 3]
```diff
@@ -52,7 +52,11 @@ Their instances may be updated via the :ref:`Command Line Interface <cli-availab
...
+is thread-safe and each thread will receive its own instance of the ``Dataset`` being accessed. Even though it is
```
Comment (awholegunch) [inline cdap-docs/developers-manual/source/building-blocks/workers.rst]: and that each thread will...

File: cdap-docs/developers-manual/source/building-blocks/workers.rst [hunk 4]
```diff
@@ -52,7 +52,11 @@ Their instances may be updated via the :ref:`Command Line Interface <cli-availab
...
+necessary to use ``getDataset`` method of ``DatasetContext`` to get a ``Dataset`` instance inside a ``TxRunnable``,
+this operation is performance optimized with the help of a Caching logic underneath::
```
Comment (awholegunch) [inline cdap-docs/developers-manual/source/building-blocks/workers.rst]: Though it is necessary to use the ``getDataset`` method of ``DatasetContext`` to access a ``Dataset`` instance inside a ``TxRunnable``, this operation is performance-optimized with the help of internal caching logic. In this example, the Dataset *tableName* is accessed from within a ``TxRunnable``::

**Comment by awholegunch** [comment]
I re-wrote the last portion, as I didn't think that the last point about caching logic flowed into the example very clearly.

**Comment by gokulavasan** [comment]
@awholegunch Please review

**Comment by awholegunch** [comment]
LGTM.

**Pull request merged**